ivadomed/scripts/automate_training.py

```diff
@@ -123,74 +129,496 @@ def test_worker(config):
+            all the first elements, then all the second elements, etc.
+
+    Returns:
+        list, dict: A list of configuration dictionaries, modified by the hyperparameters.
+
+        .. code-block:: python
+
+            config_list = [
+                {
+                    "training_parameters": {
+                        "batch_size": 18,
+                        "loss": {"name": "DiceLoss"}
+                    },
+                    "default_model":     {
+                        "name": "Unet",
+                        "dropout_rate": 0.3,
+                        "depth": 3
+                    },
+                    "model_name": "seg_tumor_t2",
+                    "path_output": "./tmp/-loss={'name': 'DiceLoss'}"
+                },
+                {
+                    "training_parameters": {
+                        "batch_size": 18,
+                        "loss": {"name": "FocalLoss", "gamma": 0.2, "alpha": 0.5}
+                    },
+                    "default_model":     {
+                        "name": "Unet",
+                        "dropout_rate": 0.3,
+                        "depth": 3
+                    },
+                    "model_name": "seg_tumor_t2",
+                    "path_output": "./tmp/-loss={'name': 'FocalLoss', 'gamma': 0.2, 'alpha': 0.5}"
+                },
+                # etc...
+            ]
+
+    """
+    config_list = []
+    if all_combin:
+        keys = set([hyper_option.base_key for hyper_option in param_list])
+        for combination in list(itertools.combinations(param_list, len(keys))):
+            if keys_are_unique(combination):
+                new_config = copy.deepcopy(initial_config)
+                path_output = new_config["path_output"]
+                for hyper_option in combination:
+                    new_config = update_dict(new_config, hyper_option.option, hyper_option.base_key)
+                    path_output = path_output + hyper_option.name
+                new_config["path_output"] = path_output
+                config_list.append(new_config)
+    elif multi_params:
+        base_keys = get_base_keys(param_list)
+        base_key_dict = {key: [] for key in base_keys}
+        for hyper_option in param_list:
+            base_key_dict[hyper_option.base_key].append(hyper_option)
+        max_length = np.min([len(base_key_dict[base_key]) for base_key in base_key_dict.keys()])
+        for i in range(0, max_length):
+            new_config = copy.deepcopy(initial_config)
+            path_output = new_config["path_output"]
+            for key in base_key_dict.keys():
+                hyper_option = base_key_dict[key][i]
+                new_config = update_dict(new_config, hyper_option.option, hyper_option.base_key)
+                path_output = path_output + hyper_option.name
+            new_config["path_output"] = path_output
+            config_list.append(new_config)
```

Comment:
could maybe be refactored? (see previous comment)

Reply:
I see what you mean, I think though they're different enough that we would only be able to factor out a few lines (since the iteration is different). Like, we could refactor out:

```
new_config = update_dict(new_config, hyper_option.option, hyper_option.base_key)
path_output = path_output + hyper_option.name
```

but I don't think that would be worth a new function?